### DISCUSSION TITLE
ES8388 failing to initialise

### DISCUSSION THREAD

**Discussion by bront32** [marcel-licence/esp32_midi_sampler#92] [General]
This is my first try at using the Audio Kit and I received the following:
rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:1344
load:0x40078000,len:13836
load:0x40080400,len:3608
entry 0x400805f0

esp32_basic_synth  Copyright (c) 2022  Marcel Licence
This program comes with ABSOLUTELY NO WARRANTY;
This is free software, and you are welcome to redistribute it
under certain conditions; 
Initialize Synth Module
Initialize I2S Module
Connect to ES8388 codec... 0x00: 0x00
0x01: 0x00
Failed!
0x00: 0x00
0x01: 0x00
Failed!

I tried swapping the ES8388_ADDR define line without success.

Anyone have a suggestion where to start looking?

**Comment by marcel-licence** [comment]
Hi,

there are some different pin configuration possible listed in the board-file:
https://github.com/marcel-licence/ML_SynthTools/blob/main/src/boards/board_audio_kit_es8388.h

**Reply by bront32** [reply]
Thanks Marcel, I got a bit further with I2C == 3 and I2S == 3 for my ESP32 Audio Kit V2.2 A247 board.  I now get...

esp32_basic_synth  Copyright (c) 2022  Marcel Licence
This program comes with ABSOLUTELY NO WARRANTY;
This is free software, and you are welcome to redistribute it
under certain conditions; 
Initialize Synth Module
Initialize I2S Module
Connect to ES8388 codec... E (978) i2c: i2c_set_pin(875): scl gpio number error
0x00: 0x00

I cannot find where that error message comes from.  Any ideas?

**Reply by bront32** [reply]
I2c,I2S set to  3,3 seems to be correct as the specification sheet sent from the seller matches the type 3 board spec sheet you linked to in the board header file.
Why the "scl gpio number error"?
I would love to see a video of your coding workflow.  Are you using the espressif ADF?  or the pschatzmann arduino-audiokit?

**Reply by marcel-licence** [reply]
Thank you for your feedback. Good question why this error occurs. I've never seen it before. I would like to add some debug output to show which ports are used. 

I didn't used the espressif ADF yet nor the pschatzmann arduino-audiokit. At the moment I tried to focus only on the ideas and audible results =)

**Comment by bront32** [comment]
I will get back to debugging soon but I can tell you it fails after “Serial.printf("0x00: 0x%02x\n", ES8388_ReadReg(ES8388_CONTROL1));” completes in es8388_begin().  The first 2 Serial.println() I put in are output ok but I never get the 3rd one after after the CONTROL2 read.

 

bool ES8388_begin(int sda, int scl, uint32_t frequency)

{

    bool ok = Wire.begin(sda, scl, frequency);

    Serial.print("wire.begin: ");    Serial.println(ok);  //*******************************

    // Reset all registers, readback default as sanity check

    //ok &= WriteReg(CHIP_AUDIO_RS, 0x123);

    delay(100);

 

    Serial.printf("0x00: 0x%02x\n", ES8388_ReadReg(ES8388_CONTROL1));

        Serial.println("control 1 read OK ");  //*******************************I get this OK

 

    Serial.printf("0x01: 0x%02x\n", ES8388_ReadReg(ES8388_CONTROL2));

        Serial.println("control 2 read OK ");  //*******************************I don’t get this

 

    ES8388_WriteReg(ES8388_CONTROL1, 1 << 7); /* do reset! */

    ES8388_WriteReg(ES8388_CONTROL1, 0x06);

    ES8388_WriteReg(ES8388_CONTROL2, 0x50);

 

    ok &= (0x06 == ES8388_ReadReg(ES8388_CONTROL1));

    ok &= (0x50 == ES8388_ReadReg(ES8388_CONTROL2));

    return ok;

}
